plugins/module_utils/ndi.py

```diff
@@ -0,0 +1,464 @@
+            if len(message) > 0:
+                result["messages"] = message
+        data = obj.get("value")["data"]
+        if len(data) > 0:
+            result["data"] = data
+        return result
+
+    def query_unhealthy_resources(self, ig_name, site_name, compliance_epoch_id):
+        result = {}
+        ig_base_path = self.event_insight_group_path.format(ig_name, site_name)
+        path = "{0}/{1}/eventUnhealthyResources?%24epochId={2}".format(ig_base_path, self.compliance_path, compliance_epoch_id)
+        objs = self.query_data(path)
+        for obj in objs:
+            result[obj.get("bucket")] = {"count": obj.get("count"), "total": obj.get("total")}
+        return result
+
+    def query_pcvs(self, ig_name):
+        pcvs_path = '{0}/{1}/prechangeAnalysis?$sort=-analysisSubmissionTime'.format(self.config_ig_path, ig_name)
+        obj = self.nd.query_obj(pcvs_path, prefix=self.prefix)
+        return obj['value']['data']
+
+    def query_pcv(self, ig_name, site_name, pcv_name):
+        pcv_results = self.query_pcvs(ig_name)
+        if pcv_name is not None and site_name is not None:
+            site_id = self.get_site_id(ig_name, site_name, prefix=self.prefix)
+            pcv_path = '{0}/{1}/fabric/{2}/prechangeAnalysis'.format(self.config_ig_path, ig_name, site_name)
+            pcv_result = self.get_pre_change_result(pcv_results, pcv_name, site_id, pcv_path, prefix=self.prefix)
+        else:
+            self.nd.fail_json(msg="site name and prechange validation job name are required")
+        return pcv_result
+
+    def is_json(self, myjson):
+        try:
+            json.loads(myjson)
+        except ValueError:
+            return False
+        return True
+
+    def load(self, fh, chunk_size=1024):
+        depth = 0
+        in_str = False
+        items = []
+        buffer = ""
+
+        while True:
+            chunk = fh.read(chunk_size)
+            if len(chunk) == 0:
+                break
+            i = 0
+            while i < len(chunk):
+                c = chunk[i]
+                buffer += c
+
+                if c == '"':
+                    in_str = not in_str
+                elif c == '[':
+                    if not in_str:
+                        depth += 1
+                elif c == ']':
+                    if not in_str:
+                        depth -= 1
+                elif c == '\\':
+                    buffer += c[i + 1]
+                    i += 1
+
+                if depth == 0:
+                    if len(buffer.strip()) > 0:
+                        j = json.loads(buffer)
+                        if not isinstance(j, list):
+                            raise AssertionError("")
+                        items += j
+                    buffer = ""
+
+                i += 1
+
+        if depth != 0:
+            raise AssertionError("Error in loading input json")
+        return items
+
+    def get_aci_class(self, prefix):
```

Comment:
Have you considered using a dictionary instead of the long if/elif?

Reply:
Will try to improve when refactor codebase